src/test/coll.rs

```diff
@@ -908,6 +901,43 @@ async fn typed_replace_one() {
         .unwrap();
 
     let result = coll.find_one(doc! { "x": 2 }, None).await.unwrap().unwrap();
-    let result: UserType = from_document(result).unwrap();
     assert_eq!(result, replacement);
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn typed_returns() {
```

Comment:
The previously written methods already test that `find`, `find_one`, and `find_one_and_replace` deserialize properly so I just added tests for `find_one_and_update` and `find_one_and_delete`.

Reply:
I think it would still be worth having tests that use the collection type without a parameter (i.e. with T = Document implicit) to make sure that things will still work fine for users.

Reply:
Is it necessary to add new tests for that? It seems that functionality is already covered by the above tests/crud spec tests.

Reply:
Good point!